api: add +required markers to scheduling API

Ignoramuss opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Add missing `+required` markers to fields in the scheduling API group to satisfy the `optionalorrequired` linter rule.

#### Which issue(s) this PR fixes:

Part of #134671

#### Special notes for your reviewer:

This adds `// +required` markers to:
- `PriorityClass.Value` in v1, v1alpha1, v1beta1
- `PriorityClassList.Items` in v1, v1alpha1, v1beta1
- `WorkloadList.Items` in v1alpha1

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```release-note

```

Ignoramuss committed:
api: add +required markers to scheduling API

Add missing +required markers to fields in the scheduling API group
to satisfy the optionalorrequired linter rule:
- PriorityClass.Value in v1, v1alpha1, v1beta1
- PriorityClassList.Items in v1, v1alpha1, v1beta1
- WorkloadList.Items in v1alpha1

Part of #134671

siyuanfoundation commented:
/ok-to-test

/cc @JoelSpeed 

/triage accepted

JoelSpeed commented:
@Ignoramuss Is this the complete set of missing optional/required markers for this API group? The other PRs we are stacking are demonstrating setting up the configuration, we may want to do that here as well

Ignoramuss committed:
Update generated.proto files with +required markers

Ignoramuss committed:
api: enable optionalorrequired linter for scheduling API

Enable the optionalorrequired linter rule and add exception patterns
for all other API groups except scheduling. This ensures the scheduling
API group is validated for proper +optional/+required markers.

Part of #134671

Ignoramuss commented:
`/retest`

Ignoramuss commented:
@JoelSpeed sure, this is the complete set for the scheduling API group. 7 fields total across v1, v1alpha1, and v1beta1 (PriorityClass.Value, PriorityClassList.Items, and WorkloadList.Items in v1alpha1).

I've also added the configuration now. Enabled the optionalorrequired rule in kube-api-linter.yaml and added exception patterns for the other API groups in exceptions.yaml (with scheduling excluded so it gets validated). The golangci yaml files are regenerated from the template.

Let me know if anything else is needed!

Ignoramuss committed:
api: fix scheduling API optional/required markers

- Change PriorityClass.Value from +required to +optional since
  there is no validation requiring it and no defaulting (0 is valid)
- Remove +required from *List.Items fields as List types should
  not need optional/required markers on their Items field

Ignoramuss committed:
api: regenerate openapi and lint config for scheduling changes

JoelSpeed commented on staging/src/k8s.io/api/scheduling/v1/types.go:
```diff
@@ -37,6 +37,7 @@ type PriorityClass struct {
 
 	// value represents the integer value of this priority class. This is the actual priority that pods
 	// receive when they have the name of this class in their pod spec.
+	// +optional
```
I can't see any defaulting for this field, nor can I see any validation that expects this field to be non-zero, so this does appear to be optional from an end user perspective

JoelSpeed reviewed (commented):
/lgtm

Thanks @Ignoramuss 

/label api-review

Ignoramuss commented:
Thanks @JoelSpeed . 

/assign @liggitt  acording to suggestion from k8s-ci-robot
